vignettes/streaming-async.Rmd

```diff
@@ -61,7 +61,76 @@ chat$chat_async("How's your day going?") %...>% print()
 #> I'm just a computer program, so I don't have feelings, but I'm here to help you with any questions you have.
 ```
 
-TODO: Shiny example
+#### Shiny example
+
+The following shiny app uses `chat_async` to query the OpenAI API. The server function shows a loading element (spinner) while it is waiting for the API to return the answer. Then the recent response is displayed.
+The `%...>%` operator from the `promises` package is used to push the asynchronous chat response and render it, once it is ready. It also hides the loading element.
+
+```
+library(shiny)
+library(bslib)
+library(elmer)
+library(promises)
+library(shinyjs)
+
+# Initialize chat
+chat <- chat_openai(
```

Comment:
Definitely should not put the chat object at this scope, should be inside the observeEvent